tests/test_cli.py

```diff
@@ -428,31 +430,37 @@ def test_all_species_genetic_maps_help(self):
 
 class TestWriteCitations(unittest.TestCase):
     """
-    Make sure all models can write citation information.
+    Make sure all engines, models, etc. can write citation information.
     """
-    def test_model(self):
-        contig = stdpopsim.Contig()
-        species = stdpopsim.get_species("homsap")
-        model = species.get_model("ooa_3")
-        engine = stdpopsim.get_default_engine()
-        stdout, stderr = capture_output(
-                cli.write_citations, engine, model, contig)
-        self.assertEqual(len(stdout), 0)
-        # TODO Parse out the output for the model and check that the text is
-        # in there.
-        self.assertGreater(len(stderr), 0)
-
-    def test_genetic_map(self):
-        species = stdpopsim.get_species("homsap")
-        contig = species.get_contig("chr22", genetic_map="HapmapII_GRCh37")
-        model = stdpopsim.PiecewiseConstantSize(species.population_size)
-        engine = stdpopsim.get_default_engine()
-        stdout, stderr = capture_output(
-                cli.write_citations, engine, model, contig)
-        self.assertEqual(len(stdout), 0)
-        # TODO Parse out the output for the model and check that the text is
-        # in there.
-        self.assertGreater(len(stderr), 0)
+    def test_citations(self):
+        for engine, species in itertools.product(
+                stdpopsim.all_engines(), stdpopsim.all_species()):
+            genetic_maps = copy.copy(species.genetic_maps)
+            if len(genetic_maps) == 0:
+                genetic_maps = [stdpopsim.GeneticMap(species.id)]
+            for genetic_map, model in itertools.product(
+                    genetic_maps, species.models):
+                contig = species.get_contig(
+                        species.genome.chromosomes[0].id,
+                        genetic_map=genetic_map.name)
+                stdout, stderr = capture_output(
+                    cli.write_citations, engine, model, contig)
```

Comment:
I haven't run your code here, but #214 doesn't alter the output streams other than appending a line to the end telling the user where the bibtex was written. So I think the only thing needed to make it compatible is to add in a `None` flag to the `write_citations` function. I.e. 

```py
bibtex=None
stdout, stderr = capture_output(
                    cli.write_citations, engine, model, contig, bibtex)
```

which is the same thing as in the [PR](https://github.com/popgensims/stdpopsim/blob/35b39a19f4abece25ab14093a7bb54bfba432d4c/tests/test_cli.py#L471-L472). It's still a WIP, but I don't see that part of it changing!